admin/class-yoast-notification-center.php

```diff
@@ -317,19 +318,23 @@ public function add_notification( Yoast_Notification $notification ) {
 		}
 
 		// Add to list.
-		$this->notifications[] = $notification;
+		$this->notifications[ $user_id ][] = $notification;
 	}
 
 	/**
-	 * Get the notification by ID.
+	 * Get the notification by ID and user ID.
 	 *
 	 * @param string $notification_id The ID of the notification to search for.
+	 * @param string $user_id         The ID of the user.
 	 *
 	 * @return null|Yoast_Notification
 	 */
-	public function get_notification_by_id( $notification_id ) {
+	public function get_notification_by_id( $notification_id, $user_id = null ) {
+		$user_id = ( ! is_null( $user_id ) ? $user_id : get_current_user_id() );
```

Comment:
I would wrap `get_current_user_id` in its own method and maybe you can pass the user_id as argument. 

```php
public function get_user_id( $user_id ) {
     if( $user_id ) {
        return $user_id;
     } 

     return get_current_user_id();
}
```